utils/python/CIME/compare_test_results.py

```diff
@@ -0,0 +1,86 @@
+import CIME.compare_namelists, CIME.simple_compare
+from CIME.utils import expect
+from CIME.XML.machines import Machines
+from CIME.test_status import *
+from CIME.hist_utils import compare_baseline
+from CIME.case import Case
+
+import os, glob
+
+_MACHINE = Machines()
+
+###############################################################################
+def compare_history(testcase_dir_for_test, baseline_name):
+###############################################################################
+    with Case(testcase_dir_for_test) as case:
+        baseline_full_dir = os.path.join(case.get_value("BASELINE_ROOT"), case.get_value("COMPILER"), baseline_name, case.get_value("CASEBASEID"))
```

Comment:
Looking at my current baselines, I see directory names like "ERS.f09_g16.I1850CLM45CN.melvin_gnu", so the compiler is clearly included in the name and therefore runs with different compilers would not conflict; so maybe adding compiler to the path is not necessary. This was a change we made in ACME a long time ago back when we were working with an ancient version of CIME, so it's possible it's not necessary anymore. For now, I can do an if/else based on model here and in bless_test_results. I don't think anywhere else is impacted.